tests/foreman/api/test_activationkey.py

```diff
@@ -42,26 +39,12 @@ def test_positive_create_2(self, max_content_hosts):
         @Feature: ActivationKey
 
         """
-        try:
-            attrs = entities.ActivationKey(
-                unlimited_content_hosts=False,
-                max_content_hosts=max_content_hosts,
-            ).create_json()
-        except HTTPError as err:
-            self.fail(err)
-        # Assert that it defaults to limited content host...
-        self.assertFalse(
-            attrs['unlimited_content_hosts'],
-            u"Unlimited content hosts is {0}".format(
-                attrs['unlimited_content_hosts'])
-        )
-        # ...and matches value passed
-        self.assertEqual(
-            attrs['max_content_hosts'],
-            max_content_hosts,
-            u"Max content hosts values don't match: {0} != {1}".format(
-                attrs['max_content_hosts'], max_content_hosts)
-        )
+        act_key = entities.ActivationKey(
+            max_content_hosts=max_content_hosts,
+            unlimited_content_hosts=False,
+        ).create()
```

Comment:
I'll let @elyezer have the final word on this. I've generally been leaving try/except blocks out of the API tests. A test fails when an assertion fails, and a test errors when an exception is thrown. The idea behind the try/except blocks is that we can catch errors and re-throw them as failures because whatever went wrong probably occurred server-side and should have been caught by an assertion in the first place. But in my experience, that's not always true - sometimes an error is due to a server-side error and sometimes an error is due to our client libraries. So rather than trying to be super smart about the exact meaning of a failure versus error, I've just been letting the code be as simple as possible.